lib/src/analyzers/lint_analyzer/rules/rules_list/prefer_correct_edge_insets_constructor/validator.dart

```diff
@@ -0,0 +1,191 @@
+part of 'prefer_correct_edge_insets_constructor_rule.dart';
+
+class _Validator {
+  final _exceptions = <InstanceCreationExpression, String>{};
+
+  _Validator();
+
+  Map<InstanceCreationExpression, String> get expressions => _exceptions;
+
+  void validate(Map<InstanceCreationExpression, EdgeInsetsData> expressions) {
+    for (final expression in expressions.entries) {
+      final data = expression.value;
+
+      if (data.constructorName == _constructorNameFromSTEB ||
```

Comment:
Consider using a switch statement here, looks more readable (especially, since multiple if blocks can't be executed)

Reply:
Done